mono/mini/mini-windows.c

```diff
@@ -383,14 +383,15 @@ mono_setup_thread_context(DWORD thread_id, MonoContext *mono_context)
 	handle = OpenThread (THREAD_ALL_ACCESS, FALSE, thread_id);
 	g_assert (handle);
 
-	context.ContextFlags = CONTEXT_INTEGER | CONTEXT_CONTROL;
+	context.ContextFlags = CONTEXT_INTEGER | CONTEXT_FLOATING_POINT | CONTEXT_CONTROL;
 
 	if (!GetThreadContext (handle, &context)) {
 		CloseHandle (handle);
 		return FALSE;
 	}
 
 	g_assert (context.ContextFlags & CONTEXT_INTEGER);
```

Comment:
These asserts are not needed.

Reply:
Just completed current implementation with additional change, but I agree they are rather unnecessary, since we are asserting something we don't have control.

Reply:
I've never seen these considered outputs, just inputs.
I guess I'll check now.

Reply:
Just looked at the documentation and not clear, but I suspect that it could be output parameter since you could ask for something not supported by the implementation, and then that won't be included in the CONTEXT on return and you can check the flag. Maybe we could just kept the asserts since they have been around for quite some time.

Reply:
Guess it works similar to newer GetXStateFeaturesMask/SetXStateFeaturesMask  API's.

Reply:
I read the code and it is input-only. The asserts do nothing.
Which is what I always figured anyway.
*Part* of the ContextFlags field is output, these parts:
CONTEXT_EXCEPTION_REPORTING;
CONTEXT_EXCEPTION_ACTIVE
CONTEXT_SERVICE_ACTIVE

but the ones you assert are never written by GetThreadContext.
Those parts are always available.

Yes you could use the XState APIs to get ymm, on processors that support it.
Requires Windows 7SP1 or so.
That API is a little weird, but would be the way to go, if the codegen can put object pointers into ymm/zmm etc.

Reply:
OK, these asserts been around forever, but if confirmed not written by GetThreadContext, then I agree we should drop them.